File: 백준/Gold/1644. 소수의 연속합/소수의 연속합.py
import sys

input = sys.stdin.readline


def prime_list(number):
    sieve = [True] * (number + 1)

    m = int(number**0.5)
    for i in range(2, m + 1):
        if sieve[i] is True:
            for j in range(i + i, number + 1, i):
                sieve[j] = False

    return [i for i in range(2, n + 1) if sieve[i]]


n = int(input())
primes = prime_list(n)

answer = 0

for i in range(len(primes) - 1):
    s = primes[i]
    j = i + 1
    while j < len(primes):
        s += primes[j]
        if n == s:
            answer += 1
            break
        elif n < s:
            break
        j += 1

if len(primes) != 0 and primes[-1] == n:
    answer += 1

print(answer)

File: 백준/Gold/1644. 소수의 연속합/소수의 연속합.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Main {

  static StringBuilder sb = new StringBuilder();
  static int n;

  public static void main(String[] args) throws IOException {
    setUp();

    int[] primeList = getPrimeList();
    int left = 0;
    int right = 0;
    int sum = primeList[0];
    int answer = 0;
    while (right < primeList.length - 1) {
      if (sum <= n) {
        if (sum == n) {
          answer++;
        }
        right++;
        sum += primeList[right];
      } else {
        sum -= primeList[left];
        left++;
      }
    }

    sb.append(answer);

    output();
  }

  private static int[] getPrimeList() {
    boolean[] sieve = new boolean[n + 1];
    Arrays.fill(sieve, true);
    List<Integer> list = new ArrayList<>();

    for (int i = 2; i < n + 1; i++) {
      if (!sieve[i]) {
        continue;
      }

      list.add(i);
      for (int j = i + i; j < n + 1; j += i) {
        sieve[j] = false;
      }
    }
    list.add(0);

    int[] result = new int[list.size()];
    for (int i = 0; i < list.size(); i++) {
      result[i] = list.get(i);
    }

    return result;
  }


  private static void setUp() throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    n = Integer.parseInt(br.readLine());
  }

  private static void output() throws IOException {
    BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    bw.write(sb.toString());
    bw.flush();
    bw.close();
  }

}